PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -454,33 +457,33 @@ class PurchasesTests: XCTestCase {
         expect(self.backend.getSubscriberCallCount).toEventually(equal(1))
     }
 
-    func testDelegateIsCalledForRandomPurchaseSuccess() {
+    func testDelegateIsCalledForRandomPurchaseSuccess() throws {
         setupPurchases()
 
-        let customerInfo = CustomerInfo(testData: emptyCustomerInfoData)
-        self.backend.postReceiptCustomerInfo = customerInfo
+        let customerInfo = try CustomerInfo(data: emptyCustomerInfoData)
+        self.backend.postReceiptResult = .success(customerInfo)
 
         let product = MockSK1Product(mockProductIdentifier: "product")
         let payment = SKPayment(product: product)
 
-        let customerInfoBeforePurchase = CustomerInfo(testData: [
+        let customerInfoBeforePurchase = try CustomerInfo(data: [
             "request_date": "2019-08-16T10:30:42Z",
             "subscriber": [
                 "first_seen": "2019-07-17T00:05:54Z",
                 "original_app_user_id": "app_user_id",
                 "subscriptions": [:],
                 "non_subscriptions": [:]
             ]])
-        let customerInfoAfterPurchase = CustomerInfo(testData: [
+        let customerInfoAfterPurchase = try CustomerInfo(data: [
```

Comment:
Same here, why have an `Optional` around when we can make sure at initialization that the data is correct.